x/lockup/spec/03_messages.md

```diff
@@ -0,0 +1,46 @@
+<!--
+order: 3
+-->
+
+# Messages
+
+## Lock Tokens
+
+`MsgLockTokens` can be submitted by any token holder via a `MsgLockTokens` transaction.
+
+```go
+type MsgLockTokens struct {
+	Proposer   sdk.AccAddress
+	UnlockTime time.Time
```

Comment:
Instead of submitting UnlockTime, let's do submit LockDuration

Reply:
Reason is because otherwise the LockDuration is based on submitted unlockTime - the block.Timestamp of the block in which the tx made it in.  If the tx is stuck in the mempool for too long, the calculated lock duration might be shorter than you want.  You wanted a 3 month lock, but because the tx was stuck for a while, you ended up getting a 2 month, 29 day, 23 hour, and 55 minute one.  And now you don't get your 3 month lockup reward!

Reply:
modified